akvo/rsr/tests/rest/test_indicator_period_data.py

```diff
@@ -1669,3 +1670,280 @@ def test_query_audit_trail(self):
+                    }]
+                }]
+            }]).build()
+        period = project.get_period(period_start=date(2010, 1, 1))
+        period.add_update(user2, value=1, status='D')
+        # When
+        self.c.login(username=user1.email, password='password')
+        response = self.c.get(f'/rest/v1/project/{project.object.id}/results_framework/?format=json')
+        # Then
+        updates = response.data['results'][0]['indicators'][0]['periods'][0]['updates']
+        self.assertEqual(0, len(updates))
+
+    def test_get_results_with_collab(self):
+        # Given
+        org = self.create_organisation('Acme')
+        user1, user2 = (self.create_user_org(f'{u}@acme.org', org, 'Enumerators') for u in ['user1', 'user2'])
+        project = ProjectFixtureBuilder()\
+            .with_partner(org, Partnership.IATI_REPORTING_ORGANISATION)\
+            .with_results([{
+                'title': 'Result  #1',
+                'indicators': [{
+                    'title': 'Indicator #1',
+                    'periods': [{
+                        'period_start': date(2010, 1, 1),
+                        'period_end': date(2010, 12, 31),
+                    }]
+                }]
+            }]).build()
+        period = project.get_period(period_start=date(2010, 1, 1))
+        update = period.add_update(user2, value=1, status='D')
+        # When
+        self.c.login(username=user1.email, password='password')
+        with override_settings(NUFFIC_ROOT_PROJECT=project.object.id):
+            response = self.c.get(f'/rest/v1/project/{project.object.id}/results_framework/?format=json')
+        # Then
+        updates = response.data['results'][0]['indicators'][0]['periods'][0]['updates']
+        self.assertEqual(1, len(updates))
+        self.assertEqual(update.id, updates[0]['id'])
+
+    def test_list_updates_no_collab(self):
+        # Given
+        org = self.create_organisation('Acme')
+        user1, user2 = (self.create_user_org(f'{u}@acme.org', org, 'Enumerators') for u in ['user1', 'user2'])
+        project = ProjectFixtureBuilder()\
+            .with_partner(org, Partnership.IATI_REPORTING_ORGANISATION)\
+            .with_results([{
+                'title': 'Result  #1',
+                'indicators': [{
+                    'title': 'Indicator #1',
+                    'periods': [{
+                        'period_start': date(2010, 1, 1),
+                        'period_end': date(2010, 12, 31),
+                    }]
+                }]
+            }]).build()
+        period = project.get_period(period_start=date(2010, 1, 1))
+        update1 = period.add_update(user2, value=1, status='A')
+        period.add_update(user2, value=2, status='D')
+        # When
+        self.c.login(username=user1.email, password='password')
+        response = self.c.get('/rest/v1/indicator_period_data_framework/?format=json', content_type='application/json')
+        # Then
+        updates = response.data['results']
```

Comment:
Why is `['results']`? Other tests compare `updates = response.data['results'][0]['indicators'][0]['periods'][0]['updates']`

Reply:
It is testing different endpoints. One is an endpoint to get the results framework of a project and the other is an endpoint to get the list of indicator period updates